table/block_based/full_filter_data.h

```diff
@@ -0,0 +1,41 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+#include <memory>
+
+#include "table/format.h"
+
+namespace rocksdb {
+
+class FilterBitsReader;
+class FilterPolicy;
+
+// The sharable/cachable part of the full filter.
+class FullFilterData {
```

Comment:
I think this should be called FullFilterCacheData or something, since I don't see it having a use outside of CachableEntry.

Reply:
`CachableEntry` is a resource handle that refers to an object that might or might *not* be in the cache (not the greatest name, although technically not a misnomer either; see `cachable_entry.h` for more details). Bottom line is that it's quite possible that the filter data it refers to is *not* in the cache; so I would rather keep the name.

Reply:
Logically, it doesn't encapsulate any more data than the reader. It encapsulates additional implementation details related to table blocks--for block cache management purposes.

How about the name ParsedFullFilterBlock? It still holds an association with being a table block but is in a ready-to-use state.

Reply:
I like `ParsedFullFilterBlock` for a variety of reasons, one being that we can keep calling the variables `block`.